### PR TITLE
helpers: Add support for product and system_ext

### PR DESCRIPTION
**Pull request by mlehtima**
[helpers] Add support for product and system_ext. JB#55653

Signed-off-by: Matti Lehtimäki <matti.lehtimaki@jolla.com>

### CODE REVIEW

**Review by mkosola — APPROVED** [review]

File: helpers/copy_tree.sh [hunk 1]
```diff
@@ -140,29 +152,48 @@ if grep -q "%define multiple_rpms 1" "$modify_spec"; then
...
+            mkdir -p "$DEVICE"/"$TREE"
+            mv "$TREE_SPARSE"/build.prop "$DEVICE"/"$TREE"
+        fi
+
+        # Move Vendor Interface Object manifest to proper place
+        if [ -f "$TREE_SPARSE"/etc/vintf/manifest.xml ]; then
+            mkdir -p "$DEVICE"/"$TREE"/etc/vintf
+            mv "$TREE_SPARSE"/etc/vintf/manifest.xml "$DEVICE"/"$TREE"/etc/vintf/manifest.xml
         fi
     fi
```
Comment (sledges) [inline helpers/copy_tree.sh:198]: Please refactor the vintf bit, along the lines of
```suggestion
        fi
    elif [ "$TREE" = "system_ext" ]; then
        if [ -f "$TREE_SPARSE"/build.prop ]; then
            DEVICE=$(grep ro."$TREE".product.name "$TREE_SPARSE"/build.prop | cut -d '_' -f3)
            if [ -z "$DEVICE" ]; then
                DEVICE=$(grep ro.product."$TREE".name "$TREE_SPARSE"/build.prop | cut -d '_' -f3)
            fi
            if [ -z "$DEVICE" ]; then
                echo "Could not parse device name from $TREE_SPARSE/build.prop"
                exit 1
            fi
            mkdir -p "$DEVICE"/"$TREE"
            mv "$TREE_SPARSE"/build.prop "$DEVICE"/"$TREE"
        fi
    fi
    if [ "$TREE" = "product" ] || [ "$TREE" = "vendor" ] || [ "$TREE" = "system_ext" ]; then
        # Move Vendor Interface Object manifest to proper place
        if [ -f "$TREE_SPARSE"/etc/vintf/manifest.xml ]; then
                mkdir -p "$DEVICE"/"$TREE"/etc/vintf
            mv "$TREE_SPARSE"/etc/vintf/manifest.xml "$DEVICE"/"$TREE"/etc/vintf/manifest.xml
        fi
    fi
```

**Review by sledges — CHANGES_REQUESTED** [review]

File: helpers/copy_tree.sh (hunk 1 above)
Comment (mlehtima) [reply helpers/copy_tree.sh]: Applied this to simplify code.

**Review by sledges — APPROVED** [review]

File: helpers/copy_tree.sh [hunk 2]
```diff
@@ -37,6 +49,7 @@ fixup_buildprop() {
         fi
         if [ -n "$BUILD_USER" ] && [ -n "$BUILD_USER_REPLACEMENT" ]; then
             sed -i "s/$BUILD_USER/$BUILD_USER_REPLACEMENT/g" "$1"
+            sed -i "s/${BUILD_USER:0:6}/$BUILD_USER_REPLACEMENT/g" "$1"
```
Comment (sledges) [inline helpers/copy_tree.sh:52]: To keep consistency:
```suggestion
            sed -i "s/${BUILD_USER:0:6}/${BUILD_USER_REPLACEMENT:0:6}/g" "$1"
```

Comment (mlehtima) [reply helpers/copy_tree.sh]: Fixed

**Review by mkosola — APPROVED** [review]

**Pull request merged**